homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -35,51 +35,52 @@ async def _async_update_data(self) -> dict:
         locks = []
         door_windows = []
 
-        locks_list = (
-            lock
-            for lock in updates["cycle"]["data"]["device_status"]
-            if lock["type"] == "device_type.door_lock"
-        )
-
-        for lock in locks_list:
-            state = lock["status1"]
-            lock_status_str = lock["minigw_lock_status"]
-            if lock_status_str != "":
-                lock_status = int(lock_status_str, 16)
-                closed = (lock_status & 16) == 16
-                locked = (lock_status & 1) == 1
+        for device in updates["cycle"]["data"]["device_status"]:
+            state = device["status1"]
+            if device["type"] == "device_type.door_lock":
+                lock_status_str = device["minigw_lock_status"]
+                if lock_status_str == "" and "device_status.lock" in state:
+                    device["_state"] = "locked"
+                    locks.append(device)
+                    continue
+                if lock_status_str == "" and "device_status.unlock" in state:
+                    device["_state"] = "unlocked"
+                    locks.append(device)
+                    continue
+                if lock_status_str != "" and (
+                    "device_status.lock" in state or "device_status.unlock" in state
+                ):
+                    lock_status = int(lock_status_str, 16)
+                    closed = (lock_status & 16) == 16
+                    locked = (lock_status & 1) == 1
                 if closed is True and locked is True:
```

Comment:
This check needs to be bested within the above block.

Reply:
I hope I understood well